doc/source/whatsnew/v2.2.0.rst

```diff
@@ -386,6 +386,33 @@ For example:
 
     pd.date_range('2020-01-01', periods=3, freq='QE-NOV')
 
+Deprecated automatic downcasting
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Deprecated the automatic downcasting of object dtype results in a number of
+methods. These would silently change the dtype in a hard to predict manner since the
+behavior was value dependent. Additionally, pandas is moving away from silent dtype
+changes (:issue:`54710`, :issue:`54261`).
+
+These methods are:
+
+- :meth:`Series.replace` and :meth:`DataFrame.replace`
+- :meth:`DataFrame.fillna`, :meth:`Series.fillna`
+- :meth:`DataFrame.ffill`, :meth:`Series.ffill`
+- :meth:`DataFrame.bfill`, :meth:`Series.bfill`
+
+Explicitly call :meth:`DataFrame.infer_objects` to replicate the current behavior in the future.
+
+```
+result = result.infer_objects(copy=False)
```

Comment:
```suggestion
.. code-block:: ipython

    In [9]: result = result.infer_objects(copy=False)

Reply:
I would do without the `In [9]` here, given this is not mimicking a runnable snippet, just a generic line of example code